atomicapp/providers/openshift.py

```diff
@@ -571,3 +622,71 @@ def _get_config_values(self):
         return (result["providerapi"],
                 result["access_token"],
                 result["namespace"])
+
+    def extract(self, image, src, dest, update=True):
+        """
+        Extract contents of a container image from 'src' in container
+        to 'dest' in host.
+
+        Args:
+            image (str): Name of container image
+            src (str): Source path in container
+            dest (str): Destination path in host
+            update (bool): Update existing destination, if True
+        """
+        cleaned_image_name = Utils.sanitizeName(image)
+        pod_name = '{}-{}'.format(cleaned_image_name, Utils.getUniqueUUID())
+        container_name = cleaned_image_name
+
+        # Pull (if needed) image and bring up a container from it
+        # with 'sleep 3600' entrypoint, just to extract content from it
+        artifact = {
+            'apiVersion': 'v1',
+            'kind': 'Pod',
+            'metadata': {
+                'name': pod_name
+            },
+            'spec': {
+                'containers': [
+                    {
+                        'image': image,
+                        'command': [
+                            'sleep',
+                            '3600'
+                        ],
+                        'imagePullPolicy': 'IfNotPresent',
+                        'name': container_name
+                    }
+                ],
+                'restartPolicy': 'Always'
+            }
+        }
+        self.oc.deploy(self._get_url(self.namespace, 'Pod'), artifact)
+        # FIXME: Pod takes some time to come up, so we wait for 3 seconds
+        # The best solution would be to watch for the pod to come to
+        # the running state
+        time.sleep(3)
+
+        # Archive content from the container and return the binary data for
+        # for the archive
```

Comment:
One potential issue I see here is that all the contents of the web request are loaded into memory and then written out to the file. If for some awful reason we ever had a very large reply to the request then we'd probably run into trouble. Would probably be better to use a buffer and write out data as we receive it so we don't keep it all in memory. 

Since I hope we come up with a better solution than this in the future we may be able to leave it like this for now but I would want it commented at the top of the execute function as well as here.